osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+               << fileName;
+    return debugEngineCleanup(client, control, symbols);
+  }
+
+  // Get stack frames from dump
+  if (control->GetStoredEventInformation(&type,
+                                         &procID,
+                                         &threadID,
+                                         context,
+                                         sizeof(context),
+                                         &contextSize,
+                                         NULL,
+                                         0,
+                                         0) == S_OK) {
+    auto contextData = new char[kNumStackFramesToLog * contextSize];
+    symbols->SetScopeFromStoredEvent();
+    auto status =
+        control->GetContextStackTrace(context,
+                                      contextSize,
+                                      stackFrames,
+                                      ARRAYSIZE(stackFrames),
+                                      contextData,
+                                      kNumStackFramesToLog * contextSize,
+                                      contextSize,
+                                      &numFrames);
+    delete[] contextData;
+    if (status != S_OK) {
+      LOG(ERROR)
+          << "Error getting context stack trace while debugging crash dump: "
+          << fileName;
+      return debugEngineCleanup(client, control, symbols);
+    }
+  } else {
+    LOG(WARNING) << "GetStoredEventInformation failed for crash dump: "
+                 << fileName;
+    if (control->GetStackTrace(
+            0, 0, 0, stackFrames, ARRAYSIZE(stackFrames), &numFrames) != S_OK) {
+      LOG(ERROR) << "Error getting stack trace while debugging crash dump: "
+                 << fileName;
+    }
+  }
+
+  std::ostringstream stackTrace;
+  auto firstFrame = true;
+  for (unsigned long frame = 0; frame < numFrames; frame++) {
+    char name[512] = {0};
+    unsigned long long offset = 0;
+
+    if (!firstFrame)
+      stackTrace << ",";
+    firstFrame = false;
+    if (symbols->GetNameByOffset(stackFrames[frame].InstructionOffset,
+                                 name,
+                                 ARRAYSIZE(name) - 1,
+                                 NULL,
+                                 &offset) == S_OK) {
+      stackTrace << name << "+0x" << std::hex << offset;
+    }
+    stackTrace << "(0x" << std::hex << stackFrames[frame].InstructionOffset;
+    stackTrace << ")";
+  }
+  r["stack_trace"] = stackTrace.str();
+
+  // Cleanup
+  return debugEngineCleanup(client, control, symbols);
+}
+
+void processDumpFile(const char* fileName, Row& r) {
+  // Open the file
+  auto dumpFile = CreateFile(fileName,
+                             GENERIC_READ,
+                             FILE_SHARE_READ | FILE_SHARE_WRITE,
+                             NULL,
+                             OPEN_EXISTING,
+                             FILE_ATTRIBUTE_NORMAL,
+                             NULL);
+  if (dumpFile == NULL) {
```

Comment:
prefer the use of `INVALID_HANDLE_VALUE` when checking `HANDLE`s just for clarity.